src/persist-client/src/internal/machine.rs

```diff
@@ -163,22 +165,41 @@ where
         applied_ever_true
     }
 
-    pub async fn register_reader(
+    pub async fn register_leased_reader(
         &mut self,
-        reader_id: &ReaderId,
+        reader_id: &LeasedReaderId,
         lease_duration: Duration,
         heartbeat_timestamp_ms: u64,
-    ) -> (Upper<T>, ReaderState<T>) {
+    ) -> (Upper<T>, LeasedReaderState<T>) {
         let metrics = Arc::clone(&self.metrics);
         let (seqno, (shard_upper, read_cap), _maintenance) = self
             .apply_unbatched_idempotent_cmd(&metrics.cmds.register, |seqno, state| {
-                state.register_reader(reader_id, seqno, lease_duration, heartbeat_timestamp_ms)
+                state.register_leased_reader(
+                    reader_id,
+                    seqno,
+                    lease_duration,
+                    heartbeat_timestamp_ms,
+                )
             })
             .await;
         debug_assert_eq!(seqno, read_cap.seqno);
         (shard_upper, read_cap)
     }
 
+    pub async fn register_critical_reader<O: Codec64 + Default>(
+        &mut self,
+        reader_id: &CriticalReaderId,
+        opaque: &O,
```

Comment:
I kinda think we should push the `O::default()` call all the way down to StateCollections::register_critical_reader? I cant think of any reason we'd let a caller specify it, even for testing